src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.cs

```diff
@@ -856,6 +858,100 @@ public ImmutableArray<T> InsertRange(int index, ReadOnlySpan<T> items)
             return self.InsertSpanRangeInternal(index, items);
         }
 
+        /// <summary>
+        /// Removes the specified values from this list.
+        /// </summary>
+        /// <param name="items">The items to remove if matches are found in this list.</param>
+        /// <param name="equalityComparer">
+        /// The equality comparer to use in the search.
+        /// </param>
+        /// <returns>
+        /// A new list with the elements removed.
+        /// </returns>
+        public ImmutableArray<T> RemoveRange(ReadOnlySpan<T> items, IEqualityComparer<T>? equalityComparer = null)
```

Comment:
Could you update the existing `IEnumerable<T>` overload to use the new algorithm?